appcheck/appcheck.go

```diff
@@ -166,6 +176,69 @@ func (c *Client) VerifyToken(token string) (*DecodedAppCheckToken, error) {
 	return &appCheckToken, nil
 }
 
+// VerifyOneTimeToken verifies the given App Check token and consumes it, so that it cannot be consumed again.
+//
+// VerifyOneTimeToken considers an App Check token string to be valid if all the following conditions are met:
+//   - The token string is a valid RS256 JWT.
+//   - The JWT contains valid issuer (iss) and audience (aud) claims that match the issuerPrefix
+//     and projectID of the tokenVerifier.
+//   - The JWT contains a valid subject (sub) claim.
+//   - The JWT is not expired, and it has been issued some time in the past.
+//   - The JWT is signed by a Firebase App Check backend server as determined by the keySource.
+//
+// If any of the above conditions are not met, an error is returned, regardless whether the token was
+// previously consumed or not.
+//
+// This method currently only supports App Check tokens exchanged from the following attestation
+// providers:
+//
+//   - Play Integrity API
+//   - Apple App Attest
+//   - Apple DeviceCheck (DCDevice tokens)
+//   - reCAPTCHA Enterprise
+//   - reCAPTCHA v3
+//   - Custom providers
+//
+// App Check tokens exchanged from debug secrets are also supported. Calling this method on an
+// otherwise valid App Check token with an unsupported provider will cause an error to be returned.
+//
+// If the token was already consumed prior to this call, an error is returned.
+func (c *Client) VerifyOneTimeToken(token string) (*DecodedAppCheckToken, error) {
```

Comment:
See my comments inline, but this function should look similar to something like this:

```go
func (c *Client) VerifyOneTimeToken(ctx context.Context, token string) (*DecodedAppCheckToken, error) {
	decodedToken, err := c.VerifyToken(token)
	if err != nil {
		return nil, err
	}

	url := fmt.Sprintf(verifyTokenURL, c.projectID)
	reqBody := map[string]string{
		"app_check_token": token,
	}
	var resp struct {
		AlreadyConsumed bool `json:"alreadyConsumed"`
	}

	req := &internal.Request{
		Method: "POST",
		URL:    url,
		Body:   internal.NewJSONEntity(reqBody),
	}
	if _, err := c.httpClient.DoAndUnmarshal(ctx, req, &resp); err != nil {
		return nil, err
	}

	decodedToken.AlreadyConsumed = &resp.AlreadyConsumed
	return decodedToken, nil
}
```